### ISSUE TITLE
模型输出 nan

### ISSUE THREAD

**Issue by UbeCc** [THUDM/GLM-4#495]
### System Info / 系統信息

transformers: 4.43.1
python: 3.10.12
openrlhf: 0.4.0

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

基于 OpenRLHF 框架 (v0.4.0, https://github.com/OpenRLHF/OpenRLHF/releases/tag/v0.4.0) 训练 reward model，使用 glm-4-9b-chat 作为 pretrained model

```shell
deepspeed ../../openrlhf/cli/train_rm.py \
   --save_path /workspace/glm-4-9b-chat-rm \
   --save_steps -1 \
   --logging_steps 1 \
   --eval_steps -1 \
   --train_batch_size 256 \
   --micro_train_batch_size 1 \
   --pretrain /workspace/ZhipuAI/glm-4-9b-chat \
   --bf16 \
   --max_epochs 1 \
   --max_len 2048 \
   --zero_stage 3 \
   --learning_rate 5e-7 \
   --dataset ~/demo.json \
   --apply_chat_template \
   --chosen_key chosen \
   --rejected_key rejected \
   --flash_attn \
   --gradient_checkpointing
```

### Expected behavior / 期待表现

在模型推理后（经过如下代码）
```python
outputs = getattr(self, self.base_model_prefix)(
    input_ids, attention_mask=attention_mask, position_ids=position_ids
)
last_hidden_states = outputs["last_hidden_state"]
```
观察到 `logits` 全部为 nan，并且没有 `last_hidden_state` 字段
如果把 `glm-4-9b-chat` 切换为 `llama-3-8b-instruct`，保持其他配置不变，训练可以正常进行。

**Comment by zhipuch** [comment]
GPU支持bf16吗？各个包的版本符合requirements中的要求吗？

**Comment by UbeCc** [comment]
> GPU支持bf16吗？各个包的版本符合requirements中的要求吗？

数据的问题，谢谢！
